### PR TITLE
265 - Add card selection

### PR DESCRIPTION
**Pull request by jeffwa428**
**Explain the details for making this change. What existing problem does the pull request solve?**
Adds a selection mixin in order to make item/card seletable, both single/multiple selection.
Here, added selection-mixin to `ids-card` so that card can be selectable.
`ids-card/selection.html` for example.

**Related github/jira issue (required)**:
Closes https://github.com/infor-design/enterprise-wc/issues/265

**Steps necessary to review your pull request (required)**:
- Go to http://localhost:4300/ids-card/single-selection.html and http://localhost:4300/ids-card/multiple-selection.html
- Try single/multiple selection for the cards

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.

### CODE REVIEW

File: src/components/ids-card/ids-card.js [hunk 1]
```diff
@@ -42,16 +48,66 @@ export default class IdsCard extends Base {
...
+    }
+
+    return this;
+  }
+
+  #changeSelection(e) {
+    this.container.querySelector('ids-checkbox').setAttribute(attributes.CHECKED, this.cardSelected !== 'true');
+    this.setAttribute(attributes.CARD_SELECTED, this.cardSelected !== 'true');
+
+    this.triggerEvent('selectionchanged', this, {
```
Comment (tmcconechy) [inline src/components/ids-card/ids-card.js:101]: Can we add a test for this? Also could it somehow be put in the mixin code?

**Review by tmcconechy — CHANGES_REQUESTED** [review]
This is a good start.. Just some edits see inline. 

Overview:

- lets call the settings selection and selected instead
- add test for the event firing
- separate examples
- when selecting the card moves 2px because the border is added
- css changes broke the layout of cards on http://localhost:4300/
- on the multiselect one the title is in the wrong place

Screen Shot 2021-12-21 at 5 03 01 PM

File: src/components/ids-process-indicator/ids-process-step/ids-process-step.scss [hunk 2]
```diff
@@ -1,8 +1,10 @@
 @import '../../../core/ids-base';
```
[the remark this answers is not in the document]
Comment (jeffwa428) [reply src/components/ids-process-indicator/ids-process-step/ids-process-step.scss:1]: I got bunch of issues from linting. So I fixed that. I am not sure how this was pushed to master.

[the remark this answers is not in the document]
Comment (tmcconechy) [reply src/components/ids-process-indicator/ids-process-step/ids-process-step.scss:1]: Possibly you created the branch before we made a bunch of changes on another PR thats also been merged. I'll check this but it probably wasn't your fault 😄

**Review by tmcconechy — CHANGES_REQUESTED** [review]
I took a look and see a couple more improvements we could make

1. The border looks a bit different stylistically (is it 3px now with some grey in it?) I compared with https://main-enterprise.demo.design.infor.com/components/cards/example-multi-select.html

Screen Shot 2021-12-27 at 12 02 44 PM

1. The checkbox on multiple-selection doesn't seem centered vertically

1. We could try to make the example look like https://main-enterprise.demo.design.infor.com/components/cards/example-multi-select.html with ids-text for the elements - i think we could use https://main.wc.design.infor.com/ids-summary-field/ but add a vertical option to put them next to each other. Im ok with leaving this as is and this can be an idea for later tho.

1. If not the previous one we could just remove the card-header and have no contents for now and put that in for a TODO for later. As i dont think that the cards will have headers in these scenarios.

3. The c

File: src/components/ids-card/ids-card.js [hunk 3]
```diff
@@ -42,16 +48,66 @@ export default class IdsCard extends Base {
...
     `;
   }
 
+  /**
+   * Establish Internal Event Handlers
+   * @private
+   * @returns {object} The object for chaining.
+   */
+  #handleEvents() {
+    this.onEvent('click', this, (e) => {
```
Comment (danielortiz1982) [inline src/components/ids-card/ids-card.js:72]: This is a nice solution, however I would recommend adding this to it's own function. We want to keep the handleEvents clean without too much business logic.

File: src/components/ids-card/ids-card.js [hunk 4]
```diff
@@ -42,16 +48,66 @@ export default class IdsCard extends Base {
...
+      if (this.selection === 'single') {
+        const cardElements = document.querySelectorAll('ids-card[selection="single"]');
+        for (const elem of cardElements) {
```
Comment (danielortiz1982) [inline src/components/ids-card/ids-card.js:75]: Just a preference thing for me, if you spread the array you can do a forEach and shorten this statement to return the setAttribute something like, `[... cardElements].forEach( (elem) => elem.setAttribute(attributes.SELECTED, false))`. This is just a preference of mine I think it makes for a cleaner pattern.

**Review by danielortiz1982 — CHANGES_REQUESTED** [review]
Remove business logic from handleEvents, Add logic for `this.selection === 'single')` into it's own function.

**Review by tmcconechy — APPROVED** [review]

**Review by jmacaluso711 — APPROVED** [review]

**Pull request merged**